lib/neopixel/blockingneopixel.cpp

```diff
@@ -1,189 +1,193 @@
-        }    
-        wait(0.5);
-        for (int i = 0; i < pixelNum; i++)
-        {
-            showColour('n');       
-        }            
-        wait(0.5);
+// tells pixels to turn on and off n times
+void BlockingNeopixel::blinkPixels(int flashes, char colour) {
+  for (int i = 0; i < flashes; i++) {
+    for (int i = 0; i < pixelNum; i++) {
+      showColour(colour);
     }
-}
-
-//Provided an array of integers of the RGB values (in GRB order), shows those
-// colours on the pixels
-void BlockingNeopixel::writeAnyRGB(const int colour[3])
-{
-    int colourBuffer[3] = {colour[0], colour[1], colour[2]};
-    int buffer[][8] = {{}, {}, {}};
-    for(int j = 0; j< 3; j++)
-    {
-        for(int i = 0; i < 8; i++){
-            buffer[j][i] = colourBuffer[j]%2;
-            colourBuffer[j] /= 2;
-        }
+    wait(0.5);
+    for (int i = 0; i < pixelNum; i++) {
+      showColour('n');
     }
-    writeByte(buffer[0]);
-    writeByte(buffer[1]);    
-    writeByte(buffer[2]);
+    wait(0.5);
+  }
 }
 
-void BlockingNeopixel::writeAnyRGBall(const int colour[3])
-{
-    int colourBuffer[3] = {colour[0], colour[1], colour[2]};
-    int buffer[][8] = {{}, {}, {}};
-    for(int j = 0; j< 3; j++)
-    {
-        for(int i = 0; i < 8; i++){
-            buffer[j][i] = colourBuffer[j]%2;
-            colourBuffer[j] /= 2;
-        }
-    }
-
-    for (int k = 0; k < pixelNum; k++)
-    {
-        writeByte(buffer[0]);
-        writeByte(buffer[1]);    
-        writeByte(buffer[2]);
+// Provided an array of integers of the RGB values (in GRB order), shows those
+// colours on the pixels
+void BlockingNeopixel::writeAnyRGB(const int colour[3]) {
+  int colourBuffer[3] = {colour[0], colour[1], colour[2]};
+  int buffer[][8] = {{}, {}, {}};
+  for (int j = 0; j < 3; j++) {
+    for (int i = 0; i < 8; i++) {
+      buffer[j][i] = colourBuffer[j] % 2;
+      colourBuffer[j] /= 2;
     }
+  }
+  writeByte(buffer[0]);
+  writeByte(buffer[1]);
+  writeByte(buffer[2]);
 }
 
-//Flashing Green
-void BlockingNeopixel::flashGreen(int numFlashes, float delay_s)
-{
-    for (int i = 0; i < numFlashes; i++)
-    {
-        showColour('g');
-        wait(delay_s);
-        showColour('n');
-        wait(delay_s);
+void BlockingNeopixel::writeAnyRGBall(const int colour[3]) {
+  int colourBuffer[3] = {colour[0], colour[1], colour[2]};
+  int buffer[][8] = {{}, {}, {}};
+  for (int j = 0; j < 3; j++) {
+    for (int i = 0; i < 8; i++) {
+      buffer[j][i] = colourBuffer[j] % 2;
+      colourBuffer[j] /= 2;
     }
-}
+  }
 
-//show solid red on all pixels
-void BlockingNeopixel::displayRed()
-{
-    showColour('r');
+  for (int k = 0; k < pixelNum; k++) {
+    writeByte(buffer[0]);
+    writeByte(buffer[1]);
+    writeByte(buffer[2]);
+  }
 }
 
-//show solid blue on all pixels
-void BlockingNeopixel::displayBlue()
-{
-    showColour('b');
+// Flashing Green
+void BlockingNeopixel::flashGreen(int numFlashes, float delay_s) {

```

Comment:
this is fine for now, but can be abstracted into a flashcolour function to make it better